rs/registry/canister/src/mutations/node_management/do_remove_node_directly.rs

```diff
@@ -387,4 +442,187 @@ mod tests {
+        let updated_operator = get_node_operator_record(&registry, node_operator_id).unwrap();
+        assert_eq!(updated_operator.node_allowance, 2);
+    }
+
+    #[test]
+    #[should_panic(expected = "Cannot remove a node that is a member of a subnet")]
+    fn should_panic_if_removing_node_in_subnet_without_replacement() {
+        let mut registry = invariant_compliant_registry(0);
+
+        // Add nodes to the registry
+        let (mutate_request, node_ids_and_dkg_pks) = prepare_registry_with_nodes(1, 1);
+        registry.maybe_apply_mutation_internal(mutate_request.mutations);
+
+        let node_ids: Vec<NodeId> = node_ids_and_dkg_pks.keys().cloned().collect();
+        let node_operator_id =
+            PrincipalId::try_from(registry.get_node_or_panic(node_ids[0]).node_operator_id)
+                .unwrap();
+
+        // Add node operator record
+        let node_operator_record = NodeOperatorRecord {
+            node_allowance: 1,
+            ..Default::default()
+        };
+
+        registry.maybe_apply_mutation_internal(vec![insert(
+            make_node_operator_record_key(node_operator_id),
+            node_operator_record.encode_to_vec(),
+        )]);
+
+        // Create a subnet with the first node
+        let subnet_id = subnet_test_id(1000);
+        let mut subnet_list_record = registry.get_subnet_list_record();
+        let subnet_record: SubnetRecord =
+            get_invariant_compliant_subnet_record(node_ids_and_dkg_pks.keys().copied().collect());
+        registry.maybe_apply_mutation_internal(add_fake_subnet(
+            subnet_id,
+            &mut subnet_list_record,
+            subnet_record,
+            &node_ids_and_dkg_pks,
+        ));
+
+        // Attempt to remove the node without replacement
+        let payload = RemoveNodeDirectlyPayload {
+            node_id: node_ids[0],
+        };
+
+        registry.do_remove_node(payload, node_operator_id);
+    }
+
+    #[test]
+    fn should_replace_node_in_subnet_and_update_allowance() {
+        let mut registry = invariant_compliant_registry(0);
+
+        // Add nodes to the registry
+        let (mutate_request, node_ids_and_dkg_pks) = prepare_registry_with_nodes(1, 2);
+        registry.maybe_apply_mutation_internal(mutate_request.mutations);
+
+        let node_ids: Vec<NodeId> = node_ids_and_dkg_pks.keys().cloned().collect();
+        let node_operator_id =
+            PrincipalId::try_from(registry.get_node_or_panic(node_ids[0]).node_operator_id)
+                .unwrap();
+
+        // Add node operator record
+        let node_operator_record = NodeOperatorRecord {
+            node_allowance: 0,
+            ..Default::default()
+        };
+
+        registry.maybe_apply_mutation_internal(vec![insert(
+            make_node_operator_record_key(node_operator_id),
+            node_operator_record.encode_to_vec(),
+        )]);
+
+        // Create a subnet with the first node
```

Comment:
IIUC, the code includes both of the nodes in the new subnet?

Reply:
fixed